src/routes/safe/components/Apps/components/ConfirmTransactionModal.tsx

```diff
@@ -114,14 +114,11 @@ export const ConfirmTransactionModal = ({
 }: OwnProps): React.ReactElement | null => {
   const [estimatedSafeTxGas, setEstimatedSafeTxGas] = useState(0)
   const threshold = useSelector(safeThresholdSelector) || 1
-  // FIXME #issue-1848 check why this generates bugs with WalletConnect Safe app and some interactions
-  // const txRecipient: string | undefined = useMemo(() => (txs.length > 1 ? MULTI_SEND_ADDRESS : txs[0]?.to), [txs])
-  // const txData: string | undefined = useMemo(() => (txs.length > 1 ? encodeMultiSendCall(txs) : txs[0]?.data), [txs])
-  // const operation = useMemo(() => (txs.length > 1 ? DELEGATE_CALL : CALL), [txs])
-  // #issue-1848 Remove this when non multisend transactions are checked
-  const txRecipient: string | undefined = MULTI_SEND_ADDRESS
-  const txData: string | undefined = useMemo(() => encodeMultiSendCall(txs), [txs])
-  const operation = DELEGATE_CALL
+
+  const txRecipient: string | undefined = useMemo(() => (txs.length > 1 ? MULTI_SEND_ADDRESS : txs[0]?.to), [txs])
+  const txData: string | undefined = useMemo(() => (txs.length > 1 ? encodeMultiSendCall(txs) : txs[0]?.data), [txs])
+  const txValue: string | undefined = useMemo(() => (txs.length > 1 ? encodeMultiSendCall(txs) : txs[0]?.value), [txs])
```

Comment:
In case it is a multisend it should be 0 not the encoded multisend

Reply:
sorry 🤦